src/main/java/io/appform/dropwizard/actors/metrics/RMQMetricObserver.java

```diff
@@ -119,6 +146,34 @@ private MetricData getMetricDataForRedelivery(final ConsumeObserverContext conte
                 getMetricData(MetricUtil.getMetricPrefixForRedelivery(metricKeyData)));
     }
 
+    private MetricData getMetricDataPerShard(final PublishObserverContext context) {
+        val metricKeyData = MetricKeyData.builder()
```

Comment:
What happens for non shared queues ?

Reply:
there is already a method for queue level metric (getMetricData) that is exporting queue level data.

Reply:
This could potentially update same metric for unsharded queues